ros2_control_test_assets/include/ros2_control_test_assets/descriptions.hpp

```diff
@@ -453,10 +579,440 @@ const auto diffbot_urdf =
+    <inertial>
+      <mass value="50"/>
+      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
+    </inertial>
+  </link>
+  <link name="finger_right">
+    <visual>
+      <geometry>
+        <box size="0.4 0.1 1"/>
+      </geometry>
+      <origin xyz="0 0.05 0.5"/>
+      <material name="grey">
+        <color rgba="0.2 0.2 0.2 1"/>
+      </material>
+    </visual>
+    <inertial>
+      <mass value="5"/>
+      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
+    </inertial>
+  </link>
+  <link name="finger_left">
+    <visual>
+      <geometry>
+        <box size="0.4 0.1 1"/>
+      </geometry>
+      <origin xyz="0 0.05 0.5"/>
+      <material name="grey">
+        <color rgba="0.2 0.2 0.2 1"/>
+      </material>
+    </visual>
+    <inertial>
+      <mass value="5"/>
+      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
+    </inertial>
+  </link>
+  <joint name="world_to_base" type="fixed">
+    <origin rpy="0 0 0" xyz="0 0 0"/>
+    <parent link="world"/>
+    <child link="base"/>
+  </joint>
+  <joint name="right_finger_joint" type="prismatic">
+    <axis xyz="0 1 0"/>
+    <origin xyz="0.0 -0.48 1" rpy="0.0 0.0 0.0"/>
+    <parent link="base"/>
+    <child link="finger_right"/>
+    <limit effort="1000.0" lower="0" upper="0.38" velocity="10"/>
+  </joint>
+)";
+
+const auto gripper_hardware_resources_no_command_if =
+  R"(
+  <ros2_control name="TestGripper" type="system">
+    <joint name="right_finger_joint">
+      <command_interface name="effort"/>
+      <state_interface name="position"/>
+      <state_interface name="velocity"/>
+      <state_interface name="effort"/>
+    </joint>
+    <joint name="left_finger_joint">
+      <state_interface name="position"/>
+      <state_interface name="velocity"/>
+    </joint>
+  </ros2_control>
+  )";
+
+const auto gripper_hardware_resources_mimic_true_no_command_if =
+  R"(
+  <ros2_control name="TestGripper" type="system">
+    <joint name="right_finger_joint">
+      <command_interface name="effort"/>
+      <state_interface name="position"/>
+      <state_interface name="velocity"/>
+      <state_interface name="effort"/>
+    </joint>
+    <joint name="left_finger_joint" mimic="true">
+      <state_interface name="position"/>
+      <state_interface name="velocity"/>
+    </joint>
+  </ros2_control>
+  )";
+
+// TODO(christophfroehlich) delete deprecated config test
+const auto gripper_hardware_resources_mimic_deprecated =
+  R"(
+  <ros2_control name="TestGripper" type="system">
+    <joint name="right_finger_joint">
+      <command_interface name="effort"/>
+      <state_interface name="position"/>
+      <state_interface name="velocity"/>
+      <state_interface name="effort"/>
+    </joint>
+    <joint name="left_finger_joint">
+      <param name="mimic">right_finger_joint</param>
+      <param name="multiplier">2</param>
+      <param name="offset">1</param>
+      <command_interface name="position"/>
```

Comment:
@christophfroehlich I don't think it makes sense to have the command interfaces for mimic joints.
What's your opinion on this?

Reply:
this is legacy code, see here:
https://github.com/ros-controls/gazebo_ros2_control/blob/074617a61db57e06c33cb4a3a92a75864bb67a39/gazebo_ros2_control_demos/urdf/test_gripper_mimic_joint.xacro.urdf#L81-L88

Reply:
I don't think it makes sense to have a command interface for a mimic joint right, if it is mimicking then it most probably is a passive joint, and passive joints shouldn't have command interfaces right?

Reply:
yes and no. Currently it mimics the same interface as the mimicked joint (if mimicked joint has effort it has automatically a mimic-effort behavior), no need for the command_interface there I guess.
I tried to improve that with [this old PR](https://github.com/ros-controls/gazebo_ros2_control/pull/111) but nowadays I think we only should mimic position at all, because I think this is the way the [URDF was designed for](http://wiki.ros.org/urdf/XML/joint)?

Reply:
I agree with you. Usually, it makes sense to have the position or velocity (I believe velocity would be nothing but the velocity of it's mimicked joint times the multiplier). Having effort here doesn't make much sense, even if it has it should report 0 in the states.

What do you think?

Reply:
if velocity or acceleration is mimicked ideally, one actually has a position-mimic with initial value offset.
Problem with gazebo classic at least is that the [ODE is broken with a kinematic chain with position interface](https://github.com/ros-controls/gazebo_ros2_control/issues/240).
So I don't see the point of mimicking anything except position, but with the drawback of the mentioned issue.

Nevertheless, this is not part of this PR. I just copied over the xml from gazebo_ros2_control demos, and we should initially support but deprecate this. After merge of this PR I update the mentioned PR above with gazebo_ros2_control.